File: load_model.js
const { PythonShell } = require('python-shell');
const { TfidfVectorizer, MultinomialNB } = require('scikit-learn'); // Assuming you're using scikit-learn for training

// Load the trained model from Python
PythonShell.run('load_model.py', null, (err, result) => {
    if (err) throw err;
    const [classifier, vectorizer] = result;
    
    // Now you can use 'classifier' and 'vectorizer' for prediction
    // Example usage:
    const review = "This is a test review";
    const numericalReview = vectorizer.transform([review]);
    const prediction = classifier.predict(numericalReview);
    console.log("Prediction:", prediction[0]);
});

File: load_model.py
import pickle

# Load the trained model
with open('trained_model.pkl', 'rb') as f:
    classifier, vectorizer = pickle.load(f)

# Return the model objects
print(classifier, vectorizer)

from sklearn.externals import joblib

# Assuming 'classifier' and 'vectorizer' are your trained model objects
joblib.dump(classifier, 'classifier.joblib')
joblib.dump(vectorizer, 'vectorizer.joblib')
